File: hidden-api/src/main/java/hidden/DexFileHidden.java
package hidden;

import android.os.Build;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;

import java.io.FileNotFoundException;

import dalvik.system.DexFile;
import dev.rikka.tools.refine.RefineAs;

@RefineAs(DexFile.class)
public class DexFileHidden {

  @RequiresApi(Build.VERSION_CODES.P)
  public static @NonNull DexFileHidden.OptimizationInfo getDexFileOptimizationInfo(
    @NonNull String fileName, @NonNull String instructionSet) throws FileNotFoundException {
    throw new RuntimeException("Stub");
  }

  @RequiresApi(Build.VERSION_CODES.P)
  public static final class OptimizationInfo {

    public @NonNull String getStatus() {
      throw new RuntimeException("Stub");
    }

    public @NonNull String getReason() {
      throw new RuntimeException("Stub");
    }
  }
}

File: hidden-api/src/main/java/hidden/DexFileHidden.kt
package hidden

import android.os.Build
import androidx.annotation.RequiresApi
import dalvik.system.DexFile
import dev.rikka.tools.refine.RefineAs
import java.io.FileNotFoundException

@RefineAs(DexFile::class)
class DexFileHidden {

  companion object {
    @RequiresApi(Build.VERSION_CODES.P)
    @JvmStatic
    @Throws(FileNotFoundException::class)
    fun getDexFileOptimizationInfo(
      fileName: String,
      instructionSet: String
    ): OptimizationInfo {
      throw RuntimeException("Stub")
    }
  }

  @RequiresApi(Build.VERSION_CODES.P)
  class OptimizationInfo {

    fun getStatus(): String {
      throw RuntimeException("Stub")
    }

    fun getReason(): String {
      throw RuntimeException("Stub")
    }
  }
}